payment/controllers.go

```diff
@@ -327,6 +328,48 @@ func GetOrderCreds(service *Service) handlers.AppHandler {
 	})
 }
 
+// GetOrderCredsByID is the handler for fetching order credentials by an item id
+func GetOrderCredsByID(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		orderID := chi.URLParam(r, "orderID")
+		itemID := chi.URLParam(r, "itemID")
+		if orderID == "" || !govalidator.IsUUIDv4(orderID) {
```

Comment:
#304 <- suggestions to get around having input validation directly within the app handlers.